connector/github/github.go

```diff
@@ -134,15 +134,14 @@ type githubConnector struct {
 }
 
 func (c *githubConnector) oauth2Config(scopes connector.Scopes) *oauth2.Config {
-	var githubScopes []string
-	if scopes.Groups {
-		githubScopes = []string{scopeEmail, scopeOrgs}
-	} else {
-		githubScopes = []string{scopeEmail}
+	// The GitHub API requires requires read:org scope to ensure users are members
+	// of orgs and teams provided in configs.
+	githubScopes := []string{scopeEmail}
+	if len(c.orgs) > 0 || c.org != "" {
```

Comment:
Instead of relying on `groups` scope, use `orgs`/`org` to determine whether `read:org` scope is needed. If neither are populated in the config, we only make `/user` and `/user/emails` API calls, which require `user:email` scope.